pkg/monitors/docker/docker.go

```diff
@@ -166,6 +167,11 @@ func (m *Monitor) Configure(conf *Config) error {
 // we aren't doing something every second across all containers, but only
 // something once every metric interval.
 func (m *Monitor) fetchStats(container dockerContainer, labelMap map[string]string, envMap map[string]string, enhancedMetricsConfig EnhancedMetricsConfig) {
+	if _, ok := m.containers[container.ID]; !ok {
```

Comment:
This is not thread-safe since maps cannot be modified and read from in separate goroutines without some kind of mutex protecting them.